javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java

```diff
@@ -2401,6 +2449,150 @@ public static List<? extends ExpressionTree> caseTreeGetExpressions(CaseTree cas
+    try {
+      return (ExpressionTree) CASETREE_GETGUARD.invoke(caseTree);
+    } catch (IllegalAccessException | InvocationTargetException e) {
+      throw new BugInCF("TreeUtils.caseTreeGetGuard: reflection failed for tree: %s", caseTree, e);
+    }
+  }
+
+  /**
+   * Returns whether {@code tree} is a {@code DefaultCaseLabelTree}.
+   *
+   * @param tree a tree to check
+   * @return whether {@code tree} is a {@code DefaultCaseLabelTree}
+   */
+  public static boolean isDefaultCaseLabelTree(Tree tree) {
+    return tree.getKind().name().contentEquals("DEFAULT_CASE_LABEL");
+  }
+
+  /**
+   * Returns whether {@code tree} is a {@code ConstantCaseLabelTree}.
+   *
+   * @param tree a tree to check
+   * @return whether {@code tree} is a {@code ConstantCaseLabelTree}
+   */
+  public static boolean isConstantCaseLabelTree(Tree tree) {
+    return tree.getKind().name().contentEquals("CONSTANT_CASE_LABEL");
+  }
+
+  /**
+   * Wrapper around {@code ConstantCaseLabelTree#getConstantExpression}.
+   *
+   * @param constantCaseLabelTree a ConstantCaseLabelTree tree
+   * @return the expression in the {@code constantCaseLabelTree}
+   */
+  public static ExpressionTree constantCaseLabelTreeGetConstantExpression(
+      Tree constantCaseLabelTree) {
+    if (!atLeastJava21) {
+      throw new BugInCF(
+          "TreeUtils.constantCaseLabelTreeGetConstantExpression: requires at least Java 21");
+    }
+    assert CONSTANTCASELABELTREE_GETCONSTANTEXPRESSION != null
+        : "@AssumeAssertion(nullness): tested atLeastJava21";
+    try {
+      @NonNull ExpressionTree ret =
+          (ExpressionTree)
+              CONSTANTCASELABELTREE_GETCONSTANTEXPRESSION.invoke(constantCaseLabelTree);
+      return ret;
+    } catch (IllegalAccessException | InvocationTargetException e) {
+      throw new BugInCF(
+          "TreeUtils.constantCaseLabelTreeGetConstantExpression: reflection failed for tree: %s",
+          constantCaseLabelTree, e);
+    }
+  }
+
+  /**
+   * Returns whether {@code tree} is a {@code PatternCaseLabelTree}.
+   *
+   * @param tree a tree to check
+   * @return whether {@code tree} is a {@code PatternCaseLabelTree}
+   */
+  public static boolean isPatternCaseLabelTree(Tree tree) {
+    return tree.getKind().name().contentEquals("PATTERN_CASE_LABEL");
+  }
+
+  /**
+   * Wrapper around {@code PatternCaseLabelTree#getPattern}.
+   *
+   * @param patternCaseLabelTree a PatternCaseLabelTree tree
+   * @return the {@code PatternTree} in the {@code patternCaseLabelTree}
+   */
+  public static Tree patternCaseLabelTreeGetPattern(Tree patternCaseLabelTree) {
+    if (!atLeastJava21) {
+      throw new BugInCF("TreeUtils.patternCaseLabelTreeGetPattern: requires at least Java 21");
+    }
+    assert PATTERNCASELABELTREE_GETPATTERN != null
+        : "@AssumeAssertion(nullness): tested atLeastJava21";
+    try {
+      @NonNull Tree ret = (Tree) PATTERNCASELABELTREE_GETPATTERN.invoke(patternCaseLabelTree);
```

Comment:
You removed the warning suppression here, but there needs to be one.  (This is also true on some of the other calls to invoke.)

Reply:
Sorry about that.  The suppressions were undocumented.  In the future, could you document the reason?  Thanks.